src/main/java/org/spongepowered/common/data/processor/data/entity/BatSleepingDataProcessor.java

```diff
@@ -0,0 +1,55 @@
+package org.spongepowered.common.data.processor.data.entity;
+
+import net.minecraft.entity.passive.EntityBat;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableSleepingData;
+import org.spongepowered.api.data.manipulator.mutable.entity.SleepingData;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeSleepingData;
+import org.spongepowered.common.data.processor.common.AbstractEntitySingleDataProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+import java.util.Optional;
+
+public class BatSleepingDataProcessor extends AbstractEntitySingleDataProcessor<EntityBat, Boolean, Value<Boolean>, SleepingData, ImmutableSleepingData> {
+
+    public BatSleepingDataProcessor() {
+		super(EntityBat.class, Keys.IS_SLEEPING);
+	}
+
+    @Override
```

Comment:
Formatting issues. We provide the Sponge standard one for Eclipse and IntelliJ that you are free to use.